src/pricer/pricer.ts

```diff
@@ -0,0 +1,93 @@
+import { BigNumber as BigNumberJs } from 'bignumber.js';
+import log from 'loglevel';
+import {
+  ExchangeApiName,
+  INTERMEDIATE_CURRENCY,
+  tokenToApi,
+  createExchangeApi,
+} from './utils';
+
+const getExchangeRate = async (
+  sourceCurrency: string,
+  targetCurrency: string,
+  intermediateCurrency = INTERMEDIATE_CURRENCY
+) => {
+  log.debug('RelayPricer: intermediary Currency', intermediateCurrency);
+
+  const sourceExchangeRate = await queryExchangeRate(
+    sourceCurrency,
+    intermediateCurrency
+  );
+
+  const targetExchangeRate = await queryExchangeRate(
+    targetCurrency,
+    intermediateCurrency
+  );
+
+  if (sourceExchangeRate.isEqualTo(0) || targetExchangeRate.isEqualTo(0)) {
+    const message = `Currency conversion for pair ${sourceCurrency}:${targetCurrency} not found in current exchange api`;
+    log.error(`RelayPricer exchange rates: ${message}`);
+    throw Error(message);
+  }
+
+  log.debug(
+    'RelayPricer exchange rates:',
+    sourceExchangeRate.toString(),
+    targetExchangeRate.toString()
+  );
+
+  return sourceExchangeRate.dividedBy(targetExchangeRate);
+};
+
+const queryExchangeRate = async (
+  sourceCurrency: string,
+  targetCurrency: string
+): Promise<BigNumberJs> => {
+  if (sourceCurrency === targetCurrency) {
+    return BigNumberJs(1);
+  }
+
+  const exchangeApis = tokenToApi[sourceCurrency.toUpperCase()];
+
+  if (!exchangeApis) {
+    throw Error(`Token ${sourceCurrency} does not have an API to query`);
+  }
+
+  log.debug(`${exchangeApis.toString()} found for token ${sourceCurrency}`);
+
+  return queryExchangeApis(exchangeApis, sourceCurrency, targetCurrency);
+};
+
+const queryExchangeApis = async (
+  exchangesApi: ExchangeApiName[],
+  sourceCurrency: string,
+  targetCurrency: string
+): Promise<BigNumberJs> => {
+  for (const api of exchangesApi) {
+    const exchangeApi = createExchangeApi(api);
+
+    const exchangeRate = await exchangeApi
+      .queryExchangeRate(sourceCurrency, targetCurrency)
```

Comment:
this will throw without catching if no api is found using `createExchangeApi`, which is what I meant in https://github.com/rsksmart/rif-relay-client/pull/85/files#r1163974138